src/node/extension/extension.ts

```diff
@@ -388,15 +386,31 @@ function startSession(config: any): StartSessionResult {
 		}
 	}
 
+	let determineDebugTypeP: Promise<string|null>;
+	if (config.request === 'attach' && typeof config.processId === 'string' && config.processId.trim() === '${command:PickProcess}') {
+		determineDebugTypeP = pickProcessAndDetermineDebugType(config).then(debugType => {
+			if (debugType) {
+				config.processId = undefined;
+				config.port = debugType === 'node2' ? INSPECTOR_PORT_DEFAULT : LEGACY_PORT_DEFAULT;
+			}
+
+			return debugType;
+		});
+	} else {
+		determineDebugTypeP = determineDebugType(config);
```

Comment:
Why is the behaviour of "determineDebugType" so different from "pickProcessAndDetermineDebugType"? I tried to use `"processId": "12345"` instead of `"processId": "${command:PickProcess}"` in my launch config (where 12345 is the ID of a node 8 process that I could have picked via the picker as well) but this fails because "determineDebugType" falls back to "legacy" without a good reason.